oneflow/core/framework/mutable_attr_map.h

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#ifndef ONEFLOW_CORE_FRAMEWORK_CACHED_ATTR_MAP_H_
+#define ONEFLOW_CORE_FRAMEWORK_CACHED_ATTR_MAP_H_
+
+#include "oneflow/core/common/util.h"
+#include "oneflow/core/common/small_vector.h"
+#include "oneflow/core/common/throw.h"
+#include "oneflow/core/framework/attr_map.h"
+#include "oneflow/core/framework/attr_value.h"
+#include "oneflow/core/framework/attr_value_accessor.h"
+#include "oneflow/core/framework/ordered_string_list.h"
+#include "oneflow/core/framework/user_op_attr.pb.h"
+#include "oneflow/core/operator/op_conf.pb.h"
+
+namespace oneflow {
+
+class MutableAttrMap {
+ public:
+  OF_DISALLOW_COPY_AND_MOVE(MutableAttrMap);
+
+  explicit MutableAttrMap(const std::vector<std::string>& attr_names)
```

Comment:
这里可以改成
```suggestion
  explicit MutableAttrMap(std::vector<std::string> attr_names)
```
然后 39 行把 attr_name std::move 一下。这样在 attr_names 是右值（绝大部分情况）下可以不拷贝

Reply:
也还是一样要拷贝，除非把OrderedStringList的构造也改一下，目前看意义不大，因为这个MutableAttrMap只初始化一次

Reply:
> 除非把OrderedStringList的构造也改一下

确实。好的